src/lib/Component/Registry.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Component;
+
+class Registry
+{
+    protected $components = [];
+
+    public function addComponent(string $group, string $serviceId, Renderable $component): void
+    {
+        $this->components[$group][$serviceId] = $component;
+    }
+
+    public function getComponents($group): array
```

Comment:
Can you add a docblock with proper return type? IDE users will benefit from that.